assets/js/data/types.d.ts

```diff
@@ -0,0 +1,8 @@
+export type ResponseError = {
```

Comment:
This is a type alias (`type Foo = …`), generally [interfaces are preferred](https://www.typescriptlang.org/docs/handbook/advanced-types.html#interfaces-vs-type-aliases). Is there a reason you used an interface here?

```suggestion
export interface ResponseError {
```

Reply:
Oops nope that was unintentional.

Reply:
done in ae1b185